cirq-ionq/cirq_ionq/results.py

```diff
@@ -26,7 +26,8 @@ class QPUResult:
     def __init__(
         self, counts: Dict[int, int], num_qubits: int, measurement_dict: Dict[str, Sequence[int]]
     ):
-        self._counts = counts
+        # Guaranteed iter ordering is required for conversion to Cirq results.
+        self._counts = collections.OrderedDict(counts)
```

Comment:
If feels strange to me that this is taking a dict, which has no ordering constraints, and then converting it to an ordered dict.  I think if the ordering is necessary, shouldn't the counts parameter have that contract?

Reply:
note that python 3.6 dicts are insertion ordered in cpython, but this is an implementation detail.  in 3.7 this becomes the standard.  So since Cirq still supports 3.6 we need to use OrderedDict.

I think you want to move the OrderedDict up to the Job results method.